Automattic/wp-calypso
# `/read/subscriptions`: Subscriptions mark follows stale on exit

Closes https://github.com/Automattic/wp-calypso/issues/78486
Closes https://github.com/Automattic/wp-calypso/issues/78377
Related to https://github.com/Automattic/wp-calypso/issues/78373

## Before
Note how the unsubscribed from site still appears in the list of subscribed-to sites, when the user is back to the "Following" page.
![Screen Capture on 2023-06-21 at 17-46-42](https://github.com/Automattic/wp-calypso/assets/2019970/53417313-b4e2-40cc-a6fa-02fa3ab04d34)


## After
Note how the unsubscribed from site does not appear in the list of subscribed-to sites, when the user is back to the "Following" page.
![Screen Capture on 2023-06-21 at 17-42-23](https://github.com/Automattic/wp-calypso/assets/2019970/e365ae9a-7c3a-4bd6-8d36-5612302b81dd)


## Proposed Changes

- In this pull request, I've implemented the new `READER_FOLLOWS_MARK_AS_STALE` Redux action. This minor adjustment ensures that the 'follows' state in the reader's Redux store is marked as stale upon unmount. The change helps keep the Redux store consistent as the user navigates, addressing an area where the subscription manager wasn't previously synchronizing with the reader's Redux store.
- To make manual testing of the above easier, I went a step further, and already changed where the "Manage" button in the Following page points to, and change the button's label from "Following" to "Subscriptions". All of that behind the `reader/subscription-management` feature flag.

## Testing Instructions



* Go to: http://calypso.localhost:3000/read
* Click at "Manage" (Subscriptions) on the right-hand side of the page. This should take you to the site subscriptions management page.
* Unsubscribe from one of the sites in your subscriptions list. When you do so, note the title of the site, you will need it for the next step.
* Next click at "Following" at the Calypso sidebar. 
* Once the Following page loads, you should see a list of your site subscriptions on the right side of that page. Use the search box, of that subscriptions list, and type in the title of the site you just unsubscribed from. The site should not be present in the list.

## Pre-merge Checklist
